### ISSUE TITLE
How to import BIRT code into eclipse

### ISSUE THREAD

**Issue by SteveSchafer-Innovent** [eclipse/birt#628]
I have not been successful at importing the BIRT code into eclipse without there being errors.  I've tried several methods of importing, including importing existing project at the root level, importing existing project for each individual project, and importing existing maven project at the root level, with varying results.  I'd like to know if anyone has been able to get the code imported without any errors and specifically how to do it.

**Comment by patric-r** [comment]
+1

Same here.
Ideally we would have a document describing how to set up a correct and working build environment within eclipse.
Building BIRT within Eclipse is an important prerequisite of doing changes, fixing bugs and issuing PRs
(I'm wondering if the main developers are  building BIRT solely with maven?).

IMHO, this would be much more important than other things (e.g. having a new logo/website).

**Comment by ruspl-afed** [comment]
@Flugtiger mentioned oomph setup script. I don't know if he still plans to submit it but this would be very helpful.

**Comment by patric-r** [comment]
A working oomph script would be helpful.

@ruspl-afed How does your BIRT development environment currently look like?

**Comment by ruspl-afed** [comment]
What I did:

(should be covered by oomph)
1) imported all project as "General" from cloned repo
2) switched target to `/org.eclipse.birt.target/org.eclipse.birt.target.target` 
3) switched workspace preference "Missed API Baseline" to warning

(plan to contribute it after tests will be fixed for master)

4) added lifycycle mapping ignorance to `/org.eclipse.birt.report.viewer/pom.xml`
5) changed VM to 1.8 for nl fragments

**Comment by patric-r** [comment]
Thanks. Which Java version/Eclipse SDK release needs to be currently used?
(Having all those information in a single document would be really helpful to the community)

**Comment by ruspl-afed** [comment]
Java 8 is important at the moment, but there is a plan to upgrade to Java 11.
I'm using the latest released Eclipse for Committers package for development.
The dependencies are specified in the target.

**Comment by SteveSchafer-Innovent** [comment]
Here's what I did. 
1. general / import existing project - select root of cloned repo.  If I do not check "Search for nested projects", I get a tree view which doesn't show packages and no errors.  If I check "Search for nested projects", it tells me a couple of projects cannot be imported because they already exist.  Then I end up with a lot of projects and 154 errors, a lot of which are lifecycle errors.
3. I changed the lifecycle error to a warning in the workspace because I didn't want to make any changes to the repo.
4. That gets me down to 143 errors, most of which are about projects that cannot be built until their prerequisites are built.  The others are various packages and/or classes that cannot be resolved.  Then after a few minutes (while I'm typing this), it decides to rebuild on its own and then I have 53 errors, most of which are "Access restriction ... is not API ...", which I can change to warnings.  After that I'm down to 17 errors which is about as low as I've been able to get it.
I think that at least some of the unresolved classes and packages may be due misconfigured java build path (e.g. no source folder), java packages no longer supported (javax.xml.bind), and J2SE-1.5 execution environment.  All of these can be fixed in due course.
I do have one really weird error where it's obviously trying to unzip a pom.xml file.  It seems to be caused by an unpack goal in the maven-dependency-plugin in birt-runtime-test.  No idea why it's doing that.
I also have "An API baseline has not been set for the current workspace".  I've tried setting it to org.eclipse.birt.target.target but that gives me "There is a possible API baseline mismatch since none of the workspace projects is in the baseline",  so that probably wan't the right thing to do.  
Anyway sorry for the long-winded comment but if anyone sees anything I'm obviously doing wrong, please let me know.

**Comment by ruspl-afed** [comment]
> I also have "An API baseline has not been set for the current workspace". I've tried setting it to org.eclipse.birt.target.target but that gives me "There is a possible API baseline mismatch since none of the workspace projects is in the baseline", so that probably wan't the right thing to do.

We don't have API baseline defined yet, but we need to create one that points to the latest officially released BIRT. @wimjongman do we know the git tag and the p2 repo URL of the latest official BIRT release?

**Comment by patric-r** [comment]
I still cannot get it to work. I used the most recent eclipse-for-commiters package, like @ruspl-afed said.
I did it like @SteveSchafer-Innovent said, using "search nested projects" on the git repo.

I have more than 20000 compile errors (e.g. org.eclipse.emf cannot be resolved - although I installled EMF, com.lowagie cannot be resolved - but when I click on the imports, the source file is opened. This really looks like a configuration error.
Any help is appreciated.

IMHO, if BIRT want to encourage other developers, the project really needs a document how to setup a proper build/development environment. Currently, it is very frustrating.

**Comment by wimjongman** [comment]
> We don't have API baseline defined yet, but we need to create one that points to the latest officially released BIRT. @wimjongman do we know the git tag and the p2 repo URL of the latest official BIRT release?

I see this release directory R-R1-4.8.0-201806261756. There is no tag for it but I expect it is commit 8e9ce5f6a29cc01ca41dee4dd6c17d3a34f19944. 

So tag BIRT_4_8_0_Release_201806261756 must be made out of commit BIRT_4_7_0_Release_201706222054

**Comment by wimjongman** [comment]
I don't have the authority to do that. I expect this can only be done by project leads.

**Comment by ruspl-afed** [comment]
> I have more than 20000 compile errors (e.g. org.eclipse.emf cannot be resolved - although I installled EMF, com.lowagie cannot be resolved - but when I click on the imports, the source file is opened. This really looks like a configuration error.
Any help is appreciated.

What target do you use? Please try with 
> /org.eclipse.birt.target/org.eclipse.birt.target.target

**Comment by patric-r** [comment]
@ruspl-afed Thanks, 

I opened /org.eclipse.birt.target/org.eclipse.birt.target.target
and clicked on "Set as Active Target Platform".

This dropped the error count down to 1052. Good!
Most of them are maven errors of the following group (all nested projects seem to be affected):

* "Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-antrun-plugin:1.8:run (execution: generate.property.files, phase: compile)" errors 
* Plugin execution not covered by lifecycle configuration: org.eclipse.tycho:tycho-packaging-plugin:1.0.0:build-qualifier (execution: default-build-qualifier, phase: validate)
* Plugin execution not covered by lifecycle configuration: org.eclipse.tycho:tycho-packaging-plugin:1.0.0:build-qualifier-aggregator (execution: default-build-qualifier-aggregator, phase: validate)
* Plugin execution not covered by lifecycle configuration: org.eclipse.tycho:tycho-packaging-plugin:1.0.0:validate-id (execution: default-validate-id, phase: validate)
* Plugin execution not covered by lifecycle configuration: org.eclipse.tycho:tycho-packaging-plugin:1.0.0:validate-version (execution: default-validate-version, phase: validate)

Any hints are appreciated.

**Comment by wimjongman** [comment]
If you open the problems view, you can select the "..lifecycle.." error. 

It enables you to download a couple of extra plugins. If that does not solve it, it also enables you to silence the errors in the preferences. Eclipse is running its own compile anyway.

**Comment by patric-r** [comment]
Thank you so much, @wimjongman ! 

I used quick fix->Discover new m2e connectors
to resolve most dependencies:

Down to 89 errors.

But not all of them:
* Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-antrun-plugin:1.8:run (execution: generate.property.files, phase: compile)

When using Quick fix, it shows me:
No marketplace entries found to handle Execution generate.property.files, in /org.eclipse.birt.report.engine.emitter.config.nl/pom.xml in Eclipse.  Please see Help for more information.


In addition to the maven errors, I have a few others:
* Java compiler level does not match the version of the installed Java project facet.
* An API baseline has not been set for the current workspace.
* The import org.eclipse.birt.chart.viewer cannot be resolved (org.eclipse.birt.chart.viewer/WebContent/drill)
* SampleHelper cannot be resolved (org.eclipse.birt.chart.viewer/WebContent)
* The method accept(File, String) of type new FilenameFilter(){} must override a superclass method (birt-runtime-test/src/test/java/org/eclipse/birt/sdk)

**Comment by wimjongman** [comment]
Can you start to write a "getting started" document that explains all steps?

**Comment by patric-r** [comment]
@wimjongman 
I can try to do this but at first I would need to have a working development environment in order to document all steps.
It would be nice if you (or someone else) would assist me to solve the remaining issues.

**Comment by ruspl-afed** [comment]
> Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-antrun-plugin:1.8:run (execution: generate.property.files, phase: compile)

please choose "Ignore" Quick Fix for this

> An API baseline has not been set for the current workspace.

Since we don't have baseline defined at the moment let's set this error to warning in preferences

As for java version related problem I fixed a number of them in master, but perhaps not all

**Comment by patric-r** [comment]
Instead of ignoring, why can't we fix the root cause of those issues?

What about the other issues?

**Comment by wimjongman** [comment]
Go to the root of the git repo and run the build. This might resolve some of the compile issues.

mvn clean verify -DskipTests

**Comment by ruspl-afed** [comment]
> Instead of ignoring, why can't we fix the root cause of those issues?

To fix the root cause we will need to implement a maven to eclipse lifecycle mapping for this maven plugin. I think this is out of scope for this project. 

> What about the other issues?

Please update us with the list of issues you still have after building the latest master with

> mvn clean verify -DskipTests

as @wimjongman suggested

**Comment by patric-r** [comment]
`mvn clean verify -DskipTests` fails very strangely with many errors.... First of them is:

```
[ERROR] C:\Users\github\git\birt\viewer\org.eclipse.birt.axis.overlay\src\org\eclipse\birt\axis\utility\UTF8Encoder.java:[1]
[ERROR]         /*
[ERROR]         ^
[ERROR] The type java.lang.Object cannot be resolved. It is indirectly referenced from required .class files
```

**Comment by wimjongman** [comment]
Yes, it needs to be run with JDK 8. I am working on a > JDK8 build and  I am one failing test away from success. 

Please try to run the same with branch prepare_for_java11

**Comment by patric-r** [comment]
@wimjongman Thanks, running maven with a JDK 8 worked. 
The command `mvn clean verify -DskipTests `ran without any errors.

However, the situation didn't change in eclipse. Even after refreshing and doing a clean build, I still have 89 errors.

**Comment by wimjongman** [comment]
What errors are left?

**Comment by wimjongman** [comment]
We just released a new build. Please update BIRT and check the remaining errors. BIRT build must now be done with JDK11+

**Comment by patric-r** [comment]
@wimjongman mvn verify goal with JDK 11 works after pulling the newest changes.
However, eclipse now shows ~5800 errors again.

Now we are back on all those "package org.w3c.dom is accessible from more than one module: <unnamed>, java.xml" errors.

**Comment by wimjongman** [comment]
Please try Project/Clean.. Clean all projects.

**Comment by patric-r** [comment]
I already did this. I refreshed maven configuration, I closed the projects and re-opened it - the result is the same.

**Comment by wimjongman** [comment]
Ok, let me try on a fresh workspace.

**Comment by patric-r** [comment]
By the way, the error makes perfectly sense for me, as the org.eclipse.birt.report.model.tests project is also providing those classes.

The interessting question is why you don't get those errors.

**Comment by ruspl-afed** [comment]
it could be because come projects are not part of pom-based build for some reason

**Comment by wimjongman** [comment]
Did you activate the target platform in the preferences/PDE/Target platform?

**Comment by patric-r** [comment]
Yes. I even clicked now on "Reload Target Platform" but this didn't change anything.

**Comment by wimjongman** [comment]
ok, I see your problems myself now. I'm looking into it.

**Comment by wimjongman** [comment]
I am still fighting this.

The issue: "package org.w3c.dom is accessible from more than one module" is now clear to me. In java 11 there can only be one package on the classpath. Some packages are passed by the JVM and by the Manifest simultaneously. 

It means that we have to juggle the build path properties while at the same time making Tycho happy.

**Comment by srbala** [comment]
Is version 4.9 release will require minimum JDK 1.8, will work and compile with JDK 11?

**Comment by wimjongman** [comment]
Yes. We just released a patch so that it compiles with JDK11. Some bundles needed to move to JDK11 for that. So strictly speaking we are not 100% compatible with JDK8 anymore.

**Comment by patric-r** [comment]
@wimjongman going back to the topic: First of all, thanks for your commits.

Just to clarify: You're still on this issue, right?
Please let me know as soon as you're finished.
I'll be happy to test this again.

**Comment by srbala** [comment]
> Some bundles needed to move to JDK11 for that. So strictly speaking we are not 100% compatible with JDK8 anymore.

@wimjongman Both JDK8 and JDK11 are in LTS. JDK8 is supported 2024 and beyond https://adoptopenjdk.net/support.html. It would be nice and better keep JDK8 support.

Builds can have profiles for JDK8 vs JDK11

**Comment by wimjongman** [comment]
Eclipse 2020-09 and above need JDK11+. People who will hack on BIRT will see the errors that are the topic of this issue.

**Comment by srbala** [comment]
Just one more clarification. So, BIRT runtime will be compatble with JDK8 and IDE tooling will require JDK 11+?

**Comment by wimjongman** [comment]
In the short term, BIRT runtime will not be compatible with JDK8.

**Comment by patric-r** [comment]
@wimjongman I executed all the steps you described.
I have only 2 java compile errors left:

PDFTranscoder cannot be resolved to a type	PDFRendererImpl.java	/org.eclipse.birt.chart.device.pdf/src/org/eclipse/birt/chart/device/pdf	line 127	Java Problem

The import org.apache.fop cannot be resolved	PDFRendererImpl.java	/org.eclipse.birt.chart.device.pdf/src/org/eclipse/birt/chart/device/pdf	line 30	Java Problem


Additionally, I still get 78 maven errors, most of them are:

Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-antrun-plugin:3.0.0:run (execution: generate.property.files, phase: compile)

**Comment by wimjongman** [comment]
The java compile errors are caused by a missing fopxx.jar in that project. By running the mvn build the file is copied to that location. You can also add it manually.

mvn errors must be resolved by installing additional mvn plugins and/or ignoring them.

**Comment by patric-r** [comment]
@wimjongman What makes this jar special compared to all other dependent jars which work out of the box?

I don't know much about the eclipse plugin ecosystem but IMHO we should either make the build pure eclipse, or we should make it pure maven. I consider a mix of both worlds (e.g. run this maven goal first then eclipse compile succeeds) unneccessarly complex and intransparent.

**Comment by wimjongman** [comment]
Note that I am only involved in building/coding BIRT for 4 weeks. There must be a historic reason but I don't know it. Probably the fop cannot be distributed with Eclipse or there is no p2 repository serving it. The reason it is not pushed into the repo is probably because of licensing. 

If fop was consumable by p2/Tycho then it should come from a recent build of the Eclipse orbit project (which does not contain it). 
https://download.eclipse.org/tools/orbit/downloads/drops/R20210223232630/

You can help by finding a p2 repo that serves our required version of fop. If you can't find it then ask the Eclipse Orbit project if they can include our required version of fop OR figure out if we can include it in our git repo. 

Make sure to create an issue for it.

**Comment by srbala** [comment]
> 
> we should either make the build pure eclipse, or we should make it pure maven. I consider a mix of both worlds (e.g. run this maven phase/goal first then eclipse compile succeeds) unneccessarly complex and intransparent.
> 
@patric-r @wimjongman 
Future direction would import project into eclipse as maven project. In some of our projects which uses the jar files which are not in p2 are resloved using manual jar install from local file. Updating Readme.md with necessary steps to setup local environment would be nice too.

**Comment by patric-r** [comment]
@wimjongman I was unable to find a p2 repo with Apache FOP 2.3.
I don't know anything about the process but wouldn't it be possible to add Apache FOP 2.3 to an official eclipse p2 repo?

@srbala Agreed, if this is possible for Eclipse Product/Plugin development, e.g. that you can still run/debug the BIRT product within Eclipse.

**Comment by wimjongman** [comment]
> @wimjongman I was unable to find a p2 repo with Apache FOP 2.3.
> I don't know anything about the process but wouldn't it be possible to add Apache FOP 2.3 to an official eclipse p2 repo?

Yes. Eclipse has a special project, called Orbit [1], that was invented exactly for this purpose. The FAQ is here [2]. There is information on how to add libraries to the Orbit project.

[1] https://www.eclipse.org/orbit/
[2] https://wiki.eclipse.org/Orbit/FAQ#How_do_I_add_something_to_Orbit.3F

**Comment by patric-r** [comment]
https://wiki.eclipse.org/Orbit/Adding_Bundles_to_Orbit#Before_You_Do_Anything

This really looks like a very bureaucratic process just for making an ASL-licensed and in maven central available jar buildable in eclipse. Is this the only way to do it? if yes: Can anybody help?
This would be so much be more valuable than having 50 mailing list posts about a new logo which nobody might care about.

**Comment by wimjongman** [comment]
> https://wiki.eclipse.org/Orbit/Adding_Bundles_to_Orbit#Before_You_Do_Anything
> 
> This really looks like a very bureaucratic process just for making an ASL-licensed and in maven central available jar usuable during an eclipse build. Is this the only way to do it? 

Yes, it is the only way to do it. Protecting our consumers from copyright infringement comes with a price.

> if yes: Can anybody help?

You can help. Looking at others to solve issues will just lead to a staring circle. Please step up if you can spare a couple of hours. I will guide you. You can start by sending a message to the Orbit mailing list.

> An easiest as possible build process seems to be so much more valuable for the project than having 50 birt-dev mailing list posts about a new BIRT project logo which nobody might care about.

Judging by the posts, many find the logo important. Not you and me though, that is why we are on this ticket 😉

**Comment by patric-r** [comment]
@wimjongman question regarding your statement:
_"Yes, it is the only way to do it. Protecting our consumers from copyright infringement comes with a price."_

BIRT is using Apache FOP in any case since years, isn't it?
What is the (legal) difference whether we use Orbit, Maven or whatever to integrate/use/package it?
The artifact delivered to the customer is exactly the same. Am I missing something?

**Comment by srbala** [comment]
@patric-r it's possible it might using older version, any new changes/version upgrade need approval from legal

**Comment by wimjongman** [comment]
> 
> 
> @wimjongman question regarding your statement:
> _"Yes, it is the only way to do it. Protecting our consumers from copyright infringement comes with a price."_
> 
> BIRT is using Apache FOP since years, isn't it?
> What is the (legal) difference whether we use Orbit, Maven or whatever to integrate/use/package it?
> The artifact delivered to the customer is exactly the same, containing the fop jar. Am I missing something?

There is no difference and we are already cleared to use Apache FOP. However, the process of getting FOP into Orbit remains the same and we have to jump through a couple of hoops to make sure that all the legal paperwork is in order.

**Comment by wimjongman** [comment]
This one is waiting on the release of the jetty p2 repository:

https://github.com/eclipse/jetty.project/pull/6404

**Comment by patric-r** [comment]
@wimjongman that PR has now finally being merged into jetty.

**Comment by wimjongman** [comment]
Is that really the case? Did you import the branch and looked at it? I would like your feedback in that case.

**Comment by patric-r** [comment]
@wimjongman Yes, but it still lacks the info which java and eclipse SDK version we need ;)

**Comment by merks** [comment]
I've tried to write Oomph setup in the current state, as observed above, it's hard to even do this manually.

For /org.eclipse.birt.chart.device.pdf/META-INF/MANIFEST.MF  lib/fop-2.3.jar is missing.
For /org.eclipse.birt.build/META-INF/MANIFEST.MF lib/dom4j-1.6.1.jar is missing.

You can't really build with compliance set to Java 11 because it complains about APIs being visible from multiple different modules.

You need a Java 8, JDK, and you need to tools.jar on the classpath or the com.sun.jdi* things don't resolve

For  /org.eclipse.birt.data.oda.pojo.ui/META-INF/MANIFEST.MF the Fragment-Host doesn't resolve, it should probably be Fragment-Host: org.eclipse.birt.data.oda.pojo;bundle-version="[2.1.0,5.0.0)" instead.

For org.eclipse.birt.report.debug.internal.ui.launcher.util.ReportLauncherUtils.getEclipseHome() it should probably be like this
```
	public static IPath getEclipseHome() {
		return new Path(TargetPlatform.getLocation());
	}
```

For org.eclipse.birt.report.viewer.mock.ServletContextSimulator it should implement more 'newer' methods.

And of course all the Maven errors should be disabled, at least for now...

There's also the issue of Jetty dependencies and where to get those.  The ones proved by the platform/SimRel aren't all the ones that are needed...

Given that things are in a state of flux and are quite badly broken, it's not really so feasible to automate the setup process at the moment.

**Comment by ruspl-afed** [comment]
Thank you so much Ed @merks for investing your time. I was creating setup in parallel, but may be I can continue from your variant. Do you plan to create a PR?

**Comment by merks** [comment]
I'm familiar with the Gerrit workflow but I don't know how to do this with forks and pull requests.  I'll just attach the work in progress here as a zip.  It's mostly just boiler plate and the JRE task isn't much good because it only works with Java 8 right now...

[BIRT.setup.zip](https://github.com/eclipse/birt/files/7446570/BIRT.setup.zip)

**Comment by wimjongman** [comment]
Sorry about this Ed. I just pushed a big stabilization change that I did not realize you were not seeing.

**Comment by wimjongman** [comment]
This one was fixed. Thanks all for pushing this. An important step forward. 

Please use the new installer link and follow the instructions. The link is also available from the README.MD:

https://www.eclipse.org/setups/installer/?url=https://raw.githubusercontent.com/eclipse/birt/master/build/org.eclipse.birt.releng/BIRTConfiguration.setup&show=true

**Comment by patric-r** [comment]
When using this installer, I still get 21 java errors, "Access Restriction" (dom4j):

![image](https://user-images.githubusercontent.com/10393382/140561286-e0ce943b-754f-4c77-97eb-976312b25168.png)

**Comment by ruspl-afed** [comment]
Strange, I thougt it was fixed by the latest manifest adjustments
Let me see if I can rebase #678

**Comment by ruspl-afed** [comment]
> When using this installer, I still get 21 java errors, "Access Restriction" (dom4j):
> 

I just merged #678 , please try to update sources from `master` branch without re-installing

**Comment by patric-r** [comment]
@ruspl-afed Yes, this fixed the problem! All java errors are now gone.
Congratulations and thanks!

However, I had several UI freezes (maybe because of the M2?). 
And every time I restart the eclipse (clean shutdown) eclipse is going to rebuild BIRT. Why?

Next questions:
- How to run the junit tests?
- How to run the report designer?

**Comment by wimjongman** [comment]
I also saw the UI freezes.

For the report designer to run you have to create a run configuration. There is already one run configuration available, you can run from there:


![image](https://user-images.githubusercontent.com/180969/140572694-4618619b-fef3-416e-903c-1940d27e7176.png)



For the unit test to run you also have to create a run configuration. You can start finding the classes that are name AllTests and run these as JUnit plugins tests. This will create a run configuration that you will have to tweak in the Run Configuration dialog


![image](https://user-images.githubusercontent.com/180969/140573138-96b14a97-e21e-4303-900c-295d1cce6577.png)



![image](https://user-images.githubusercontent.com/180969/140573251-d8f18dcd-1a7d-4134-b9b4-7d20ffa41cbb.png)

**Comment by patric-r** [comment]
When running the "birtrun" configuration, I get:

![image](https://user-images.githubusercontent.com/10393382/140575277-b7755355-4fae-4089-bb3b-dc9a62c15639.png)

Is this a known issue=

A click on "continue" starts the report designer.
In the console I see:

```
osgi> Nov 05, 2021 9:35:53 PM org.eclipse.birt.report.model.metadata.ExtensionLoader handleError
SEVERE: Message:Duplicate extension name when adding the extension [RotatedText] to the meta-data dictionary. Error code:DUPLICATE_EXTENSION_NAME 
(along with other SWT exceptions)
```

Has something changed regarding the project explorer in BIRT?
I used to create (report) projects and organize my rptdesign files there but it doesn't seem to work like this anymore.
I can create a Report Project, but it is not shown in the project explorer.

(will test junit later)

**Comment by wimjongman** [comment]
> is this a known issue?
Yes.

In Eclipse you can have several starting points called applications. Currently, in the launch configuration, the DesignerApplication is used. You can play with that to launch BIRT in several 'modes'.


![image](https://user-images.githubusercontent.com/180969/140576547-d22116c0-5764-467a-bfdb-4a1a4ad0d86a.png)

**Comment by patric-r** [comment]
Thanks. Have you seen my 2nd question?
Somethings seems to be really broken in the designer, neither the project explorer nor the navigator will show any created project.

**Comment by ruspl-afed** [comment]
> Somethings seems to be really broken in the designer, neither the project explorer nor the navigator will show any created project.

Please create another issue to discuss this. We will need steps to reproduce. Please also check if you can see the same problem when running from pre-built BIRT binaries.

**Comment by wimjongman** [comment]
> 
> 
> Thanks. Have you seen my 2nd question? Somethings seems to be really broken in the designer, neither the project explorer nor the navigator will show any created project.

That was an answer to your second question. Change the application to see BIRT in different outfits. The one you are looking at now does not have the project explorer. https://github.com/eclipse/birt/issues/628#issuecomment-962210935

**Comment by patric-r** [comment]
@ruspl-afed: Which pre-built BIRT binaries do you mean? This one? https://download.eclipse.org/birt/downloads/drops/I-R1-4.8.0-201805221921/birt-report-designer-all-in-one-4.8.0-20180522-win32.win32.x86_64.zip 

@wimjongman 
Which "application" corresponds to the BIRT designer (= what happens when I execute the eclipse binary in above package)? I just assumed that it's the default (DesignerApplication) and then I would assume that it behaves in the same way

**Comment by ruspl-afed** [comment]
> @ruspl-afed: Which pre-built BIRT binaries do you mean? This one? https://download.eclipse.org/birt/downloads/drops/I-R1-4.8.0-201805221921/birt-report-designer-all-in-one-4.8.0-20180522-win32.win32.x86_64.zip

the binaries that you can produce with the local maven build.
please open another issue if you think that BIRT Designer has a bug, it seems to be not related with this ticket

**Comment by wimjongman** [comment]
Patrick we can continue this discussion here: https://github.com/eclipse/birt/issues/692
